src/utils/script.ts

```diff
@@ -1,15 +1,30 @@
 import { CharacterNameSchema, characterRegistry } from "@/data/characters/registry";
 import type { Script } from "@/types/script";
+import type { HomebrewCharacter } from "@/data/characters/homebrew";
 import type { CollectionEntry } from "astro:content";
 import { match } from "ts-pattern";
 
 export const parseScriptData = (script: CollectionEntry<"scripts">): Script => {
   const [meta, ...characters] = script.data;
 
-  const characterBreakdown = characters.reduce(
-    (acc, character) => {
-      const parsedCharacter = CharacterNameSchema.parse(character);
+  return {
+    id: script.id,
+    name: meta.name,
+    author: meta.author,
+    characterBreakdown: meta.almanac
+      ? homebrewCharacterBreakdown(characters as HomebrewCharacter[])
+      : standardCharacterBreakdown(characters as string[]),
+    characters: meta.almanac
+      ? (characters as HomebrewCharacter[])
+      : characters.map((character) => CharacterNameSchema.parse(character as string)),
+    scriptJsonString: JSON.stringify(script.data),
+  };
+};
 
+const standardCharacterBreakdown = (characters: string[]): Script["characterBreakdown"] => {
+  return characters.reduce(
+    (acc, character) => {
+      const parsedCharacter = CharacterNameSchema.parse(character as string);
```

Comment:
```suggestion
      const parsedCharacter = CharacterNameSchema.parse(character);
```

nit: this one shouldn't be needed